docs/design/2022-06-15-phyical_plan_explain.md

```diff
@@ -0,0 +1,684 @@
+
+#### `CoveringIndexScan`
+
+`CoveringIndexScan` is a special case of `IndexScan`. If the column(s) in Filter and the column(s) in Projection are both inside the Index in one visit, then we only need to scan the Index once, no need to scan for the Table. That is, it only scans for the Index, such a scan is called `CoveringIndexScan`.
+
+#### `TableScan`
+
+`TableScan` is different from `IndexScan` and `CoveringIndexScan`. `TableScan` only scans table data. Such a scan we called `TableRangeScan` in the following.
+
+### Filter Concept
+
+#### `PushDown Filter`
+
+The expression passed to COP([TiKV Coprocessor](https://docs.pingcap.com/tidb/stable/tikv-overview#tikv-coprocessor))/TiKV as Selection without triggering a downgrade.
+
+#### `Downgrade Filter`
+
+`Downgrade Filter` is used after `IndexScan` downgrade.
+
+The first stage of `IndexScan` which we called `IndexRangeScan` will return the `RowID` that meets the conditions, and then TiSpark will sort and aggregate the returned Row IDs to obtain the region that needs to be scanned in the second stage of `IndexScan`——`TableRowIDScan`. After sorting and aggregating we will get the `regionTask` number. If the `regionTask` number is bigger than `downgradeThreshold`, The `TableRowIDScan` in the second will be `TableScan`.  The `StartKey` of `TableRangeScan` is the minimum `RowID` returned by `IndexRangeScan`, and the `EndKey ` of `TableRangeScan` is the minimum `RowID` returned by `IndexRangeScan`. The Filter will become `Downgrade Filter` (`Downgrade Filter` is the same as if the execution plan is `TableScan`'s Filter).
+
+> **`RegionTask`**
+>
+> For all returned `RowID`, all consecutive `RowID` in a region will be treated as a `RegionTask`.
+>
+> For example like this the `RowID` returned in the first stage are 1,3,4,5 and 1,3,4 are in the same region and 5 is in another region. Since 1 and 3,4 are not contiguous, 1 is `RegionTask`, and since 3,4 and 5 are not in a region, 3,4 is a `RegionTask` and 5 is another `RegionTask`.The `RegionTask` number will three.
+
+#### `Residual Filter`
+
+In the original design, the `Residual Filter` represents operators that cannot be pushed down to COP/TiKV. However, in the current implementation, before the construction of `DAGRequest`, it will judge whether the operators can be pushed down, and only the operators that can be downscaled will participate in the construction of `DAGRequest`, which means that all the operators in `DAGRequest` can be downscaled to COP/TiKV. The Residual Filter loses its original meaning because there are no operators in `DAGRequest` that cannot be pushed down.
```

Comment:
```suggestion
In the original design, the `Residual Filter` represents operators that cannot be pushed down to COP/TiKV. However, in the current implementation, before the construction of `DAGRequest`, it will judge whether the operators can be pushed down, and only the operators that can be pushed down will participate in the construction of `DAGRequest`, which means that all the operators in `DAGRequest` can be pushed down to COP/TiKV. The Residual Filter loses its original meaning because there are no operators in `DAGRequest` that cannot be pushed down.
```